CppCoreGuidelines.md

```diff
@@ -3294,15 +3295,17 @@ such as `string` and `vector`, that needs to do free store allocations.
 
 To compare, if we passed out all values as return values, we would something like this:
 
-    pair<istream&, string> get_string(istream& in)  // not recommended
+    struct get_string_result { istream& in; string s; };
+
+    get_string_result get_string(istream& in)  // not recommended
     {
         string s;
         in >> s;
-        return {in, move(s)};
+        return { in, move(s) };
     }
 
-    for (auto p = get_string(cin); p.first; p.second = get_string(p.first).second) {
-        // do something with p.second
+    for (auto [in, s] = get_string(cin); in; s = get_string(stream).s) {
```

Comment:
```suggestion
    for (auto [in, s] = get_string(cin); in; s = get_string(in).s) {
```

Also, I can't make suggestions on the surrounding context, but "we would something like this" on line 3296 is missing a word. I think it should be "we would have something like this".